test/metrics.bats

```diff
@@ -11,27 +11,49 @@ function teardown() {
 	cleanup_test
 }
 
-@test "metrics with default port" {
-	# start crio with default port 9090
+@test "metrics with default host and port" {
+	HOST="127.0.0.1"
 	PORT="9090"
+
+	# Start CRI-O with default host and port.
 	CONTAINER_ENABLE_METRICS=true start_crio
-	if ! port_listens "$PORT"; then
-		skip "Metrics port $PORT not listening"
+
+	# The metrics endpoint should not listen on all available interfaces.
+	if host_and_port_listens "::" $PORT; then
+		echo "Metrics endpoint should not listen on all interfaces" >&3
+		return 1
 	fi
 
-	# get metrics
-	curl -sf "http://localhost:$PORT/metrics"
+	curl -sf "http://${HOST}:${PORT}/metrics" | grep crio_operations
 }
 
-@test "metrics with random port" {
-	# start crio with custom port
+@test "metrics with custom host using localhost and random port" {
+	HOST="localhost"
 	PORT=$(free_port)
-	CONTAINER_ENABLE_METRICS=true CONTAINER_METRICS_PORT=$PORT start_crio
+
+	# Start CRI-O using a custom host of "localhost",
+	# which should be the same as "127.0.0.1", so the
+	# same as default.
+	CONTAINER_ENABLE_METRICS=true CONTAINER_METRICS_HOST=$HOST CONTAINER_METRICS_PORT=$PORT start_crio
 
 	crictl run "$TESTDATA"/container_redis.json "$TESTDATA"/sandbox_config.json
 
-	# get metrics
-	curl -sf "http://localhost:$PORT/metrics" | grep crio_operations
+	curl -sf "http://localhost:${PORT}/metrics" | grep crio_operations
+}
+
+@test "metrics with custom host using 127.0.1.1" {
+	HOST="127.0.1.1"
+	PORT="9090"
+
+	# Start CRI-O using a custom IP of "127.0.1.1".
+	CONTAINER_ENABLE_METRICS=true CONTAINER_METRICS_HOST=$HOST start_crio
+
+	if ! host_and_port_listens $HOST $PORT; then
+		echo "Metrics endpoint should listen on 127.0.1.1" >&3
+		return 1
+	fi
+
+	curl -sf "http://${HOST}:${PORT}/metrics" | grep crio_operations
 }
```

Comment:
The test failures seems to be legit: 

```
not ok 189 metrics with custom host using 127.0.1.1
# (in test file ./metrics.bats, line 53)
#   `return 1' failed
```

https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/cri-o_cri-o/7714/pull-ci-cri-o-cri-o-main-ci-fedora-integration/1751893021156184064/artifacts/fedora-integration/cri-o-gather/artifacts/testout.txt

Reply:
@saschagrunert, yeah. These have been a hit-and-miss.

I get intermittent failures locally, too. Reason? All run on the same host without isolation.

I think, I will drop some of these tests. Not sure how to fix the concurrent execution issue, aside from setting concurrency to 1 for bats.

Reply:
Done.